test/e2e/upgradetest/upgrade_test.go

```diff
@@ -114,3 +122,90 @@ func TestHealOneMemberForOldCluster(t *testing.T) {
 		t.Fatalf("failed to heal one member: %v", err)
 	}
 }
+
+// TestBackupForOldCluster tests that new backup sidecar could make backup from old cluster.
+func TestBackupForOldCluster(t *testing.T) {
+	t.Run("PV backup for old cluster", func(t *testing.T) {
+		testBackupForOldClusterWithBackupPolicy(t, e2eutil.NewPVBackupPolicy())
+	})
+	t.Run("S3 backup for old cluster", func(t *testing.T) {
+		testBackupForOldClusterWithBackupPolicy(t, e2eutil.NewS3BackupPolicy())
+	})
+}
+
+func testBackupForOldClusterWithBackupPolicy(t *testing.T, bp *spec.BackupPolicy) {
+	err := testF.CreateOperator()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		err := testF.DeleteOperator()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
+	err = k8sutil.WaitEtcdTPRReady(testF.KubeCli.CoreV1().RESTClient(), 3*time.Second, 30*time.Second, testF.KubeNS)
+	if err != nil {
+		t.Fatalf("failed to see cluster TPR get created in time: %v", err)
+	}
+
+	bp.BackupIntervalInSecond = 60 * 60 * 24 // long enough that no backup was made automatically
+	bp.CleanupBackupsOnClusterDelete = true
+	cl := e2eutil.NewCluster("upgrade-test-", 3)
+	cl = e2eutil.ClusterWithBackup(cl, bp)
+	testClus, err := e2eutil.CreateCluster(t, testF.KubeCli, testF.KubeNS, cl)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		checker := &e2eutil.StorageCheckerOptions{
+			S3Cli:    testF.S3Cli,
+			S3Bucket: testF.S3Bucket,
+		}
+		if err := e2eutil.DeleteCluster(t, testF.KubeCli, testClus, checker); err != nil {
+			t.Fatal(err)
+		}
+	}()
+	_, err = e2eutil.WaitUntilSizeReached(t, testF.KubeCli, 3, 60*time.Second, testClus)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// "latest" image operator will create sidecar with versioned image. So we need to get sidecar image.
+	oldBSImage, err := getBackupSidecarImage(k8sutil.BackupSidecarName(testClus.Metadata.Name))
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = testF.UpgradeOperator()
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println("upgraded")
+
+	// We wait for old backup sidecar to be deleted so that we won't make backup from the old version.
+	lo := metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(k8sutil.BackupSidecarLabels(testClus.Metadata.Name)).String(),
+	}
+	_, err = e2eutil.WaitPodsWithImageDeleted(testF.KubeCli, testF.KubeNS, oldBSImage, 30*time.Second, lo)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println("backup pod deleted")
+	// Since old one is deleted, we can safely select any backup pod.
+	err = e2eutil.WaitBackupPodUp(t, testF.KubeCli, testF.KubeNS, testClus.Metadata.Name, 60*time.Second)
+	if err != nil {
+		t.Fatal(err)
+	}
+	fmt.Println("backup pod running")
+	err = e2eutil.MakeBackup(testF.KubeCli, testClus.Metadata.Namespace, testClus.Metadata.Name)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func getBackupSidecarImage(sidecarName string) (string, error) {
```

Comment:
why this is specific to sidecar? getContainerImageName(podName string) // get the image of the first container in the pod.